src/sentry_core.c

```diff
@@ -1472,6 +1472,24 @@ sentry_attach_bytes_n(
         buf, buf_len, sentry__path_from_str_n(filename, filename_len)));
 }
 
+void
+sentry_clear_attachments(void)
+{
+    SENTRY_WITH_SCOPE_MUT (scope) {
+        SENTRY_WITH_OPTIONS (options) {
+            if (options->backend && options->backend->remove_attachment_func) {
```

Comment:
consider a separate `sentry_backend_t::clear_attachments_func`
- to avoid nested locking of options and scope
- to avoid several IPC messages